### PR TITLE
Analytics updates

### PR DESCRIPTION
**Pull request by OchiengPaul442**
#### Summary of Changes (What does this PR do?)

- fixed organization select issue by modifying customfield component

image

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `formatName` function for improved name formatting.
	- Enhanced conditional formatting for selected options based on specific IDs.

- **Bug Fixes**
	- Improved handling of input field values to ensure proper formatting.

- **Documentation**
	- Updated prop types to enforce required properties and improve clarity on expected structures.

### CODE REVIEW

File: platform/src/common/components/Modal/dataDownload/components/CustomFields.jsx [hunk 1]
```diff
@@ -35,47 +41,64 @@ const CustomFields = ({
...
       {field ? (
+        // Render a text input field
         <input
           className="bg-transparent text-[16px] font-medium leading-6 p-0 m-0 w-full h-auto border-none"
-          value={formatName(selectedOption.name)}
+          value={
+            id === 'organization'
+              ? selectedOption.name
+              : formatName(selectedOption.name, textFormat)
+          }
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply platform/src/common/components/Modal/dataDownload/components/CustomFields.jsx:76]: Hi @OchiengPaul442 
- please take note of this review comment.
- But also please ensure that you maintain our earlier UPPER CASE only organisation name implementation. Please do NOT change it.
cc: @Codebmk

[the remark this answers is not in the document]
Comment (OchiengPaul442) [reply platform/src/common/components/Modal/dataDownload/components/CustomFields.jsx:76]: Upper casing is still maintained @Baalmart

[the remark this answers is not in the document]
Comment (OchiengPaul442) [reply platform/src/common/components/Modal/dataDownload/components/CustomFields.jsx:76]: will push a commit on this  comment

[the remark this answers is not in the document]
Comment (Baalmart) [reply platform/src/common/components/Modal/dataDownload/components/CustomFields.jsx:76]: Thanks @OchiengPaul442

**Pull request merged**